spec/notifications/publishers/job_application_data_expiry_notification_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "rails_helper"
+
+RSpec.describe Publishers::JobApplicationDataExpiryNotification do
+  let(:organisation) { create(:school) }
+  let(:publisher) { create(:publisher) }
+  let(:vacancy) { create(:vacancy, publisher: publisher, organisation_vacancies_attributes: [{ organisation: organisation }]) }
+
+  describe "#message" do
+    subject { Notification.last.to_notification.message }
+
+    let(:data_expiration_date) { (vacancy.expires_at + 1.year).to_date }
+    let(:vacancy_applications_link) { "/organisation/jobs/#{vacancy.id}/job_applications" }
+
+    before do
+      described_class
+        .with(vacancy: vacancy, publisher: publisher)
+        .deliver(publisher)
+    end
+
+    it "returns the correct message" do
+      expect(subject).to include(vacancy.job_title)
+                     .and include(format_date(data_expiration_date))
+                     .and include(vacancy_applications_link)
+    end
+  end
+
+  describe "#timestamp" do
+    subject { Notification.last.to_notification.timestamp }
+
+    context "when the notification is delivered today" do
+      before do
+        described_class
+          .with(vacancy: vacancy, publisher: publisher)
+          .deliver(publisher)
+      end
+
+      it "returns the correct timestamp" do
+        expect(subject).to match(/Today at/)
+      end
+    end
+
+    context "when the notification was delivered yesterday" do
+      before do
+        travel_to 1.day.ago do
+          described_class
+            .with(vacancy: vacancy, publisher: publisher)
+            .deliver(publisher)
+        end
+      end
+
+      it "returns the correct timestamp" do
+        expect(subject).to match(/Yesterday at/)
```

Comment:
is using a regex necessary here? (aassuming thats what this is - looks the same as JS)

Reply:
The other notification spec checks against values given in the same way. I copied it lol

Reply:
I guess it's only checking that the timestamp has "Yesterday at" in it. I'll change it now

Reply:
I've changed these to use the include matcher instead now. Don't know why Connor used `to match` instead.